lib/index.js

```diff
@@ -1723,32 +1740,23 @@ exposeMethod('pluck');
  */
 
 Stream.prototype.pickBy = function (f) {
-
-    return this.consume(function (err, x, push, next) {
+    return this.map(objectOnly(function (x) {
         var out = {};
-        if (err) {
-            push(err);
-            next();
-        }
-        else if (x === nil) {
-            push(err, x);
-        }
-        else if (_.isObject(x)) {
-            for (var k in x) {
-                if (f(k, x[k])) {
-                    out[k] = x[k];
-                }
+        var props = [];  // cache of already seen properties
+        var obj = x;  // variable used to transverse prototype chain
+
+        // function created here so we don't create one per loop
+        var testAndAdd = function (prop) {
```

Comment:
In this case, it is an optimization b/c it creates the function once instead of once per loop iteration.  I placed a comment above it to note why this code is inconsistent with the rest of the library.